backend/src/main/java/ai/verta/modeldb/advancedService/AdvancedServiceImpl.java

```diff
@@ -234,17 +236,29 @@ public void getHydratedExperimentsByProjectId(
       mdbRoleService.validateEntityUserWithUserInfo(
           ModelDBServiceResourceTypes.PROJECT, request.getProjectId(), ModelDBServiceActions.READ);
 
-      var findExperimentResponse =
-          futureExperimentDAO
-              .findExperiments(
-                  FindExperiments.newBuilder()
-                      .setProjectId(request.getProjectId())
-                      .setPageLimit(request.getPageLimit())
-                      .setPageNumber(request.getPageNumber())
-                      .setAscending(request.getAscending())
-                      .setSortKey(request.getSortKey())
-                      .build())
-              .get();
+      FindExperiments.Response findExperimentResponse;
+      try {
+        findExperimentResponse =
+            futureExperimentDAO
+                .findExperiments(
+                    FindExperiments.newBuilder()
+                        .setProjectId(request.getProjectId())
+                        .setPageLimit(request.getPageLimit())
+                        .setPageNumber(request.getPageNumber())
+                        .setAscending(request.getAscending())
+                        .setSortKey(request.getSortKey())
+                        .build())
+                .get();
+      } catch (Exception ex) {
+        if (ex.getCause() != null && ex.getCause().getCause() != null) {
+          if (ex.getCause().getCause() instanceof InvalidArgumentException) {
+            throw (InvalidArgumentException) ex.getCause().getCause();
+          } else if (ex.getCause().getCause() instanceof UnimplementedException) {
+            throw (UnimplementedException) ex.getCause().getCause();
+          }
+        }
+        throw ex;
+      }
```

Comment:
In AdvanceServiceImpl we do not implemented InternalFuture so we have to catch and rethrow errors so it is doing here